polkadot/xcm/xcm-builder/src/barriers.rs

```diff
@@ -88,6 +88,8 @@ impl<T: Contains<MultiLocation>> ShouldExecute for AllowTopLevelPaidExecutionFro
 				_ => Err(ProcessMessageError::BadFormat),
 			})?
 			.skip_inst_while(|inst| matches!(inst, ClearOrigin))?
+			// allow setting fees mode to jit or not for use in following `BuyExecution`
+			.skip_inst_while(|inst| matches!(inst, SetFeesMode { .. }))?
```

Comment:
```suggestion
```

iiuc, I think at least the doc is not 100% correct because `BuyExecution` does not use fees_mode - it just directly decreases holding register `self.holding.try_take(fees, ...)`. `SetFeesMode` does not affect `BuyExecution` instruction.
`SetFeesMode`is useful only for instructions which ends in `fn send(..)` or `fn take_fee(..)`
_Unrelated note, `SetFeesMode { jit_withdraw: true }` will never work after instruction `ClearOrigin`._

What is the real scenario for this added `SetFeesMode` here or it is just some kind of leftover?

Reply:
yes, this is leftover, thank you, good catch!